main.go

```diff
@@ -64,7 +64,8 @@ func NewCmd() *cobra.Command {
 	cmd.Flags().StringVarP(&flags.OutputDelimiter, "output-delimiter", "o", " ",
 		"Can only be used in conjunction with the --echo flag, this specifies the delimiter used to separate the\n"+
 			DOC_INDENT+"input and output. Defaults to \" \"")
-
+	cmd.Flags().StringVarP(&flags.ShellInterpreter, "shell-interpreter", "i", "/bin/bash",
```

Comment:
Tested a few of the "supported" shells, not all of them export their own `SHELL` var. Grabbing the parent PID and looking up its `/proc/cmdline` (or the Mac equivalent) seems like a much more reliable default. I'll work on implementing that.

Reply:
Awesome! If you can make that happen it'd be sweet. Would it work over ssh though? I.e.

```
ssh hostname "echo | map 'cat /proc/\$PID/cmdline'"
```

Or whatever the equivalent would be. I think at this point we might be trying to be too accommodating. I'm ok with something more sane like `-i` defaulting to `/bin/bash`, unless an env var called `MAP_SHELL_INTERPRETER` is defined, instead of this nonsense.